crates/ruff_linter/src/rules/pyupgrade/rules/yield_in_for_loop.rs

```diff
@@ -130,10 +132,22 @@ pub(crate) fn yield_in_for_loop(checker: &Checker, stmt_for: &ast::StmtFor) {
         format!("yield from {contents}")
     };
 
-    diagnostic.set_fix(Fix::unsafe_edit(Edit::range_replacement(
-        contents,
-        stmt_for.range(),
-    )));
+    if !collect_names(value).any(|name| {
```

Comment:
Does it make sense to even raise the diagnostic in the case of a non-local or global? How would a user rewrite the code in that case to preserve the existing semantics?

Reply:
That makes sense. And this should then not raise the rule. Should we add this as an example to the `no_errors` test or is it already covered by other examples?